src/tests/test_fastod.cpp

```diff
@@ -0,0 +1,509 @@
+            config::InputTable table, config::TimeLimitSecondsType time_limit_seconds = 0u) {
+        using namespace config::names;
+
+        algos::StdParamsMap params{{kTable, table}, {kTimeLimitSeconds, time_limit_seconds}};
+
+        return algos::CreateAndLoadAlgorithm<algos::fastod::Fastod>(params);
+    }
+
+    static void TestFastod(fs::path const& dataset_path,
+                           std::vector<std::string> expected_asc_ods_str,
+                           std::vector<std::string> expected_desc_ods_str,
+                           std::vector<std::string> expected_simple_ods_str) {
+        using namespace algos::fastod;
+
+        config::InputTable table = std::make_shared<CSVParser>(dataset_path, ',', true);
+
+        auto algorithm = CreateAlgorithmInstance(table);
+        algorithm->Execute();
+
+        std::vector<AscCanonicalOD> asc_ods = algorithm->GetAscendingDependencies();
+        std::vector<DescCanonicalOD> desc_ods = algorithm->GetDescendingDependencies();
+        std::vector<SimpleCanonicalOD> simple_ods = algorithm->GetSimpleDependencies();
+
+        std::vector<std::string> asc_ods_str(asc_ods.size());
+        std::vector<std::string> desc_ods_str(desc_ods.size());
+        std::vector<std::string> simple_ods_str(simple_ods.size());
+
+        std::transform(asc_ods.cbegin(), asc_ods.cend(), asc_ods_str.begin(),
+                       [](AscCanonicalOD const& od) { return od.ToString(); });
+        std::transform(desc_ods.cbegin(), desc_ods.cend(), desc_ods_str.begin(),
+                       [](DescCanonicalOD const& od) { return od.ToString(); });
+        std::transform(simple_ods.cbegin(), simple_ods.cend(), simple_ods_str.begin(),
+                       [](SimpleCanonicalOD const& od) { return od.ToString(); });
+
+        ASSERT_EQ(asc_ods_str.size(), expected_asc_ods_str.size());
+        ASSERT_EQ(desc_ods_str.size(), expected_desc_ods_str.size());
+        ASSERT_EQ(simple_ods_str.size(), expected_simple_ods_str.size());
+
+        std::sort(asc_ods_str.begin(), asc_ods_str.end());
+        std::sort(desc_ods_str.begin(), desc_ods_str.end());
+        std::sort(simple_ods_str.begin(), simple_ods_str.end());
+        std::sort(expected_asc_ods_str.begin(), expected_asc_ods_str.end());
+        std::sort(expected_desc_ods_str.begin(), expected_desc_ods_str.end());
+        std::sort(expected_simple_ods_str.begin(), expected_simple_ods_str.end());
+
+        for (size_t i = 0; i < asc_ods_str.size(); ++i) {
+            ASSERT_EQ(asc_ods_str[i], expected_asc_ods_str[i]);
+        }
+
+        for (size_t i = 0; i < desc_ods_str.size(); ++i) {
+            ASSERT_EQ(desc_ods_str[i], expected_desc_ods_str[i]);
+        }
+
+        for (size_t i = 0; i < simple_ods_str.size(); ++i) {
+            ASSERT_EQ(simple_ods_str[i], expected_simple_ods_str[i]);
+        }
+    }
+};
+
+TEST_F(FastodTest, TestSmall1) {
+    fs::path dataset_path = fs::path(input_data_path) / "small_2x3.csv";
```

Comment:
We use csv configs from `src/tests/all_csv_configs.cpp`. You need to add configs there or use them if they already exist. Then use them in tests.

Reply:
Did it.